bots/mainbot/crusader.js

```diff
@@ -6,34 +6,97 @@ import combat from './combat.js'
 const crusader = {};
 
 crusader.takeTurn = (self) => {
-	let enemies = utilities.enemiesInRange(self);
-	self.castleTalk(SPECS.CRUSADER)
-	// if (self.moveQueue === undefined) {
-	// 	self.moveQueue = [];
-	// }
-    // if(enemies.length > 0){
-    //     for(let i = 0; i < enemies.length; ++i){
-    //     	if(enemies[i].unit === SPECS.CASTLE){
-    //     		self.log("Attacking Castle");
-    //     		return combat.attackBot(self, enemies[i]);
-	// 		}
-	// 	}
-    // }
-    // if (self.step === 0 ){
-    // 	self.log("Searching for castle on horizontal symmetric map");
-    //     self.moveQueue = movement.moveTo(self, (self.map[0].length - self.me.x), (self.me.y))
-    // }
-	// if (self.moveQueue.length !== 0) {
-	// 	let move = self.moveQueue.shift();
-	// 	self.log("moving to " + (move.x) + ', ' + (move.y));
-	// 	return self.move((move.x - self.me.x), (move.y - self.me.y));
-	// }
-    // else if(self.step !== 0 && enemies.length < 1){
-    // 	self.log("Searching for castle on vertically symmetric map");
-    //     self.moveQueue = movement.moveTo(self, (self.map[0].length - self.me.x), (self.map.length - self.me.y))
-    // }
+	self.log("crusader turn")
+    self.castleTalk(SPECS.CRUSADER)
+    if (self.horizontalCastle === undefined){
+        self.horizontalCastle = {x: (self.map[0].length - self.me.x), y: self.me.y};
+    }
+    if (self.verticalCastle === undefined){
+        self.verticalCastle = {x: self.me.x, y: (self.map.length - self.me.y)};
+    }
+    if (self.targetCastle === undefined){
+        self.targetCastle = self.horizontalCastle;
+    }
+    if(self.attacking === undefined){
+        self.attacking = false;
+    }
+    let enemies = utilities.enemiesInRange(self);
+    // If attack skip phase check
+    if (self.attacking === true){
+        utilities.log(self, "Continuing attack phase")
+        return crusader.attackPhase(self, enemies);
+    }
+    // If we aren't attacking check the phase to see if castle has said to attack
+	let phase = utilities.getCastleSignal(self);
 
+	if(phase === 1){
+        self.attacking = true;
+		return crusader.attackPhase(self, enemies);
+	}
+	else{
+		utilities.log(self, "defaulting to build phase");
+        return crusader.buildPhase(self, enemies);
+	}
+};
+
+crusader.attackPhase = (self, enemies) => {
+    utilities.log(self, "Executing Attack Phase turn");
+    if (self.moveQueue === undefined) {
+        self.moveQueue = [];
+    }
+    utilities.log(self, `Found ${enemies.length} enemies near me`)
+    if(enemies.length > 0){
+        for(let i = 0; i < enemies.length; ++i){
+            if(enemies[i].unit === SPECS.CASTLE){
+                self.log("Attacking Castle");
+            }
+            return combat.attackBot(self, enemies[i]);
```

Comment:
This appears to always attack the first bot in enemies, regardless of whether there's a castle later in the list or not. The for loop doesn't appear to ever loop, since the return statement is outside of the if block.